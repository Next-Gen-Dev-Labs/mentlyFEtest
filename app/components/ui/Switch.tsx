"use client";

import * as React from "react";

const Switch = () => (
  <div className="h-[16px] w-[32px] rounded-xl bg-white px-[2px] flex justify-end items-center">
    <div className="bg-[#1F0954] w-[12px] h-[12px] rounded-full" />
  </div>
);

export { Switch };
