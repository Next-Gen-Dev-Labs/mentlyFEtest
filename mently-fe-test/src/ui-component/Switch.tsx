

"use client";

import * as SwitchPrimitive from "@radix-ui/react-switch";
import { twMerge } from "tailwind-merge";

type SwitchProps = {
  checked: boolean;
  onChange: (checked: boolean) => void;
  disabled?: boolean;
  className?: string;
};

export default function Switch({
  checked,
  onChange,
  disabled,
  className,
}: SwitchProps) {

  const handleCheckedChange = (newChecked: boolean) => {
    onChange(newChecked);
  };

  return (
    <SwitchPrimitive.Root
      className={twMerge(
        "relative w-[50px] h-6 rounded-full bg-white data-[state=checked]:bg-white transition shrink-0 cursor-pointer",
        disabled && "opacity-50 cursor-not-allowed",
        className
      )}
      checked={checked}
      onCheckedChange={handleCheckedChange}
      disabled={disabled}
    >
      <SwitchPrimitive.Thumb className="block w-4 h-4 bg-primary rounded-full shadow-md transition-transform transform translate-x-1 data-[state=checked]:translate-x-8" />
    </SwitchPrimitive.Root>
  );
}


