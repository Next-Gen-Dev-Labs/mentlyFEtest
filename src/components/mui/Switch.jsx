'use client';
import { useState } from 'react';

export default function MySwitch({enabled,setEnabled}) {

  return (
    <label
      className="relative inline-flex items-center cursor-pointer"
      style={{
        width: '29.23px',
        height: '14.62px',
        borderRadius: '18.27px',
        transform: 'rotate(-180deg)',
      }}
    >
      <input
        type="checkbox"
        className="sr-only peer"
        checked={!enabled}
        onChange={() => setEnabled(!enabled)}
      />

      {/* Track */}
      <div
        className={`w-full h-full transition-colors duration-300`}
        style={{
          backgroundColor: !enabled ? '#FFFFFF' : '#C2C2C2',
          borderRadius: '18.27px',
          position: 'absolute',
          top: '2.19px',
          left: '0', // Keep the track within the bounds
          width: '29.23px',
          height: '14.62px',
          transform: 'rotate(-180deg)',
        }}
      />

      {/* Knob */}
      <div
        className={`absolute transition-transform duration-300`}
        style={{
          width: '11.69px',
          height: '11.69px',
          borderRadius: '50%',
          backgroundColor: !enabled ? '#1F0954' : '#FFFFFF',
          top: '3.65px',
          // Adjust the translation to move the knob within the bounds
          transform: `translateX(${enabled ? '14.62px' : '0'}) rotate(-180deg)`,
        }}
      />
    </label>
  );
}
