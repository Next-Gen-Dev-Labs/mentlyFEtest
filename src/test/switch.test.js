import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import Switch from '@/components/reusables/switch';

describe('Switch Component', () => {
  const mockToggle = jest.fn();
  const props = {
    isOn: false,
    handleToggle: mockToggle,
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly in off state', () => {
    render(<Switch {...props} />);
    
    const switchContainer = screen.getByRole('switch');
    expect(switchContainer).toBeInTheDocument();
    expect(switchContainer).toHaveClass('bg-gray-300');
    expect(switchContainer).toHaveClass('w-10');
    expect(switchContainer).toHaveClass('h-4');
    
    const switchThumb = switchContainer.firstChild;
    expect(switchThumb).toHaveClass('translate-x-0');
  });

  it('renders correctly in on state', () => {
    render(<Switch {...props} isOn={true} />);
    
    const switchContainer = screen.getByRole('switch');
    expect(switchContainer).toHaveClass('bg-blue-500');
    
    const switchThumb = switchContainer.firstChild;
    expect(switchThumb).toHaveClass('translate-x-6');
  });

  it('calls handleToggle when clicked', () => {
    render(<Switch {...props} />);
    
    const switchContainer = screen.getByRole('switch');
    fireEvent.click(switchContainer);
    
    expect(mockToggle).toHaveBeenCalledTimes(1);
  });

  it('applies correct transition classes', () => {
    render(<Switch {...props} />);
    
    const switchContainer = screen.getByRole('switch');
    const switchThumb = switchContainer.firstChild;
    
    expect(switchContainer).toHaveClass('transition-colors');
    expect(switchContainer).toHaveClass('duration-300');
    expect(switchThumb).toHaveClass('transition-transform');
    expect(switchThumb).toHaveClass('duration-300');
  });

  it('has proper accessibility attributes', () => {
    const { rerender } = render(<Switch {...props} />);
    
    const switchContainer = screen.getByRole('switch');
    expect(switchContainer).toHaveAttribute('aria-checked', 'false');
    expect(switchContainer).toHaveAttribute('tabindex', '0');
    
    // Test with on state using rerender
    rerender(<Switch {...props} isOn={true} />);
    expect(switchContainer).toHaveAttribute('aria-checked', 'true');
  });

  it('matches snapshot in off state', () => {
    const { asFragment } = render(<Switch {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot in on state', () => {
    const { asFragment } = render(<Switch {...props} isOn={true} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('responds to keyboard events', () => {
    render(<Switch {...props} />);
    const switchContainer = screen.getByRole('switch');
    
    // Test space key
    fireEvent.keyDown(switchContainer, { key: ' ' });
    expect(mockToggle).toHaveBeenCalledTimes(1);
    
    // Test enter key
    fireEvent.keyDown(switchContainer, { key: 'Enter' });
    expect(mockToggle).toHaveBeenCalledTimes(2);
  });
});