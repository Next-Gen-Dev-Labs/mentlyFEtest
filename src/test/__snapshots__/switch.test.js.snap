// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch Component matches snapshot in off state 1`] = `
<DocumentFragment>
  <div
    aria-checked="false"
    class="w-10 h-4 flex items-center bg-white rounded-full p-0.5  cursor-pointer transition-colors duration-300 bg-gray-300"
    role="switch"
    tabindex="0"
  >
    <div
      class="bg-[#1F0954] w-3 h-3 rounded-full shadow-md transform transition-transform duration-300 translate-x-0"
    />
  </div>
</DocumentFragment>
`;

exports[`Switch Component matches snapshot in on state 1`] = `
<DocumentFragment>
  <div
    aria-checked="true"
    class="w-10 h-4 flex items-center bg-white rounded-full p-0.5  cursor-pointer transition-colors duration-300 bg-blue-500"
    role="switch"
    tabindex="0"
  >
    <div
      class="bg-[#1F0954] w-3 h-3 rounded-full shadow-md transform transition-transform duration-300 translate-x-6"
    />
  </div>
</DocumentFragment>
`;
